app/actions/views/channel.js

```diff
@@ -647,15 +649,45 @@ export function loadChannelsForTeam(teamId, skipDispatch = false) {
                             });
                         }
                     }
-
-                    dispatch(batchActions(actions, 'BATCH_LOAD_CHANNELS_FOR_TEAM'));
                 }
 
                 // Fetch needed profiles from channel creators and direct channels
                 dispatch(loadSidebar(data));
 
                 dispatch(loadUnreadChannelPosts(data.channels, data.channelMembers));
             }
+
+            if (data.team) {
+                let result = await Client4.getAllGroupsAssociatedToChannelsInTeam(teamId, true);
```

Comment:
This whole block should be wrapped with its own try/catch and you should cehck for a min server version before this block.

Why?
Try/catch cause if any of this request fail they should not make the whole action to fail.
Min version cause only newer servers will have this endpoints, any other server will just fail and as the code is written the whole action will then fail, making it impoasible to complete fetching the channels.

Reply:
Done! I have wrapped it in a try/catch however I'm just not sure what to do if it errors. Currently, I just return the error because I wasn't sure what needs to be done.